crates/wasi/src/preview2/host/io.rs

```diff
@@ -124,71 +124,53 @@ impl<T: WasiView> streams::HostOutputStream for T {
 
     async fn splice(
         &mut self,
-        _dst: Resource<OutputStream>,
-        _src: Resource<InputStream>,
-        _len: u64,
+        dest: Resource<OutputStream>,
+        src: Resource<InputStream>,
+        len: u64,
     ) -> StreamResult<u64> {
-        // TODO: We can't get two streams at the same time because they both
-        // carry the exclusive lifetime of `ctx`. When [`get_many_mut`] is
-        // stabilized, that could allow us to add a `get_many_stream_mut` or
-        // so which lets us do this.
-        //
-        // [`get_many_mut`]: https://doc.rust-lang.org/stable/std/collections/hash_map/struct.HashMap.html#method.get_many_mut
-        /*
-        let s: &mut Box<dyn crate::InputStream> = ctx
-            .table_mut()
-            .get_input_stream_mut(src)
-            ?;
-        let d: &mut Box<dyn crate::OutputStream> = ctx
-            .table_mut()
-            .get_mut(dst)
-            ?;
-
-        let bytes_spliced: u64 = s.splice(&mut **d, len).await?;
-
-        Ok(bytes_spliced)
-        */
-        todo!("stream splice is not implemented")
+        let len = len.try_into().unwrap_or(usize::MAX);
+        let output = self.table_mut().get_mut(&dest)?;
+        let permit = output.check_write()?;
+        let _ = output;
```

Comment:
no longer necessary I think?

Reply:
Oh? I would imagine that NLL fixes this because `let _: &mut T = val;` AFAIK does nothing in terms of lifetime inference

Reply:
oh. yes, you're right. im not sure what error message i was trying to get around by writing it that way, but it isnt necessary. at any rate i like the way its written now.